crates/compiler/module/src/symbol.rs

```diff
@@ -1239,8 +1255,12 @@ define_builtins! {
         4 BOOL_OR: "or"
         5 BOOL_NOT: "not"
         6 BOOL_XOR: "xor"
-        7 BOOL_EQ: "isEq"
-        8 BOOL_NEQ: "isNotEq"
+        7 BOOL_NEQ: "isNotEq"
+        8 BOOL_EQ: "Eq" exposed_type=true
+        9 BOOL_IS_EQ: "isEq"
+        10 BOOL_IS_EQ_IMPL: "boolIsEq"
+        unexposed 11 BOOL_STRUCTURAL_EQ: "structuralEq"
+        unexposed 12 BOOL_STRUCTURAL_NOT_EQ: "structuralNotEq"
```

Comment:
Oh nice, is this a new feature in this macro?

Reply:
Yep! To mark functions only exposed for compiler-internal reasons. Will also be relevant for #4318